### ISSUE TITLE
why bn is not used in your network?

### ISSUE THREAD

**Issue by Aryayay** [mrharicot/monodepth#45]
same as the title

**Comment by mrharicot** [comment]
Please read the paper, page 5, section 4.1.
https://arxiv.org/pdf/1609.03677.pdf

**Comment by Aryayay** [comment]
but when I use bn in the encoder part, when train with large epochs, the model is still stable

**Comment by mrharicot** [comment]
Large epochs or large learning rates?

I am not saying it won't work.
But back when we got the first arXiv version of the paper up, batchnorm was causing more issues than not. We did use it in both the encoder and decoder at the time though iirc.

**Comment by yzcjtr** [comment]
Actually I experimented with adding BN again. From my experience, BN doesn't give any improvement, which is consistent with the paper.

**Comment by Aryayay** [comment]
@mrharicot I mean large epochs, in your paper you said that "This model was only trained for 12 epochs as it becomes unstable if trained for longer.", but with BN it can train with more epochs, and best "d1_all" I got  is 8.36, better than 9.19 in your paper.

I use resnet when training, and another problem,  in file "monodepth_nodel.py" line 142:
do_proj = tf.shape(x)[3] != num_layers or stride == 2, here do_proj always true, and I fixed the problem, the model size compressed from 685MB to 585MB

**Comment by 7LFB** [comment]
@Aryayay  can the results be reproduced when you re-run it when using batch-norm?

@mrharicot  can you list some issues when you using the batch norm in  your first version ? 

thank you!!
